IntegrationTests/Runner/Scenarios/URLSession/URLSessionScenarios.swift

```diff
@@ -142,11 +142,12 @@ class URLSessionBaseScenario: NSObject {
     }
 
     private func createInstrumentedURLSession() -> URLSession {
-        let delegate: URLSessionDelegate
+        let delegate: URLSessionDataDelegate
 
         switch setup.instrumentationMethod {
         case .directWithGlobalFirstPartyHosts:
             delegate = DDURLSessionDelegate()
+            URLSessionInstrumentation.enable(with: .init(delegateClass: DDURLSessionDelegate.self))
```

Comment:
question/ There should be no difference if using (our) `DDURLSessionDelegate` vs (their) `MyCustomDelegate`, right? If the plan is to remove `DDURLSessionDelegate`, we could start using custom delegate in integration tests right away.

Reply:
these test for compatibility reasons, we want existing customers to not break.

Which means if customers are using `DDURLSessionDelegate`, we will auto enable urlsession instrumentation.

Reply:
however, we need to find deprecation process.